Objects/object.c

```diff
@@ -2134,6 +2134,14 @@ _PyTrash_end(PyThreadState *tstate)
 }
 
 
+int
+_PyTrash_cond(PyObject *op, void *dealloc)
+{
+    PyTypeObject *tp = Py_TYPE(op);
+    return tp->tp_dealloc == (destructor)dealloc;
```

Comment:
I don't think that defining a tp type is useful here.

```suggestion
    return Py_TYPE(op)->tp_dealloc == (destructor)dealloc;
```

Reply:
Ok, updated.